include/klee/Solver/Solver.h

```diff
@@ -17,16 +17,23 @@
 #include <vector>
 
 namespace klee {
-  class ConstraintManager;
+  class ConstraintSet;
   class Expr;
   class SolverImpl;
 
+  /// Collection of meta data used by a solver
+  ///
+  struct SolverQueryMetaData {
+    /// @brief Costs for all queries issued for this state, in seconds
```

Comment:
Please drop "in seconds". To put it simply - a time span has no unit. I guess you are creating the struct here for future extensions?

Reply:
E.g., in MoKLEE you could pass a reference to the ptree node and pass that struct further down the chain instead of a complete ExecutionState.

Reply:
Right -- I guess I simply felt a comment is in place (both of us said something along the lines "I assume this is because...")

Reply:
Or you could carry a potential `Assignment` through the solver chain, so you don't need to look it up in the CexCache.